Libraries/PeriphDrivers/Source/LP/lp_me15.c

```diff
@@ -102,9 +104,94 @@ void MXC_LP_EnterShutDownMode(void)
     // Should never reach this line - device will reset on exit from shutdown mode.
 }
 
-void MXC_LP_SetOVR(mxc_lp_ovr_t ovr)
+int MXC_LP_SetOVR(mxc_lp_ovr_t ovr)
 {
-    //not supported yet
+    uint32_t current_clock, div;
+    int error;
+
+    // Ensure part is operating from internal LDO for core power
+    if (MXC_PWRSEQ->lpcn & MXC_F_PWRSEQ_LPCN_LDO_DIS) {
+        return E_BAD_STATE;
+    }
+
+    // Select the 8KHz nanoring (no guarantee 32KHz is attached) as system clock source
+    current_clock = MXC_GCR->clkctrl & MXC_F_GCR_CLKCTRL_SYSCLK_SEL;
+    if (current_clock == MXC_SYS_CLOCK_IPO) {
+        error = MXC_SYS_Clock_Select(MXC_SYS_CLOCK_INRO);
+        if (error != E_NO_ERROR) {
+            return error;
+        }
+    }
+
+    // Set flash wait state for any clock so its not to low after clock changes.
+    MXC_GCR->memctrl = (MXC_GCR->memctrl & ~(MXC_F_GCR_MEMCTRL_FWS)) |
+                       (0x5UL << MXC_F_GCR_MEMCTRL_FWS_POS);
+
+    // Set the OVR bits
+    MXC_PWRSEQ->lpcn &= ~(MXC_F_PWRSEQ_LPCN_OVR);
+    MXC_PWRSEQ->lpcn |= ovr;
```

Comment:
I'd suggest using MXC_Setfield here to ensure ovr doesn't overwrite any other field in that register or check that it's a valid selection beforehand.